src/components/membership-queue.ts

```diff
@@ -88,6 +180,8 @@ export class MembershipQueue {
             reason,
             kickUser,
             type: "leave",
+            ts: Date.now(),
+            ttl: ttl || this.opts.defaultTtl,
```

Comment:
If `ttl` is set to 0 this will pick the default.

```suggestion
            ttl: typeof ttl === 'number' ? ttl : this.opts.defaultTtl,
```

well... until `??` comes along.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator
Supported starting from Node 14...

Reply:
Look forward to that in a couple of years :+1:

Reply:
>If ttl is set to 0 this will pick the default.

This is intended as a value of 0 would mean fail the request immediately.